### ISSUE TITLE
linter: unicorn/require-post-message-target-origin false positive

### ISSUE THREAD

**Issue by tmr232** [oxc-project/oxc#10106]
### What version of Oxlint are you using?

0.16.3

### What command did you run?

`oxlint -D suspicious`

### What does your `.oxlintrc.json` config file look like?

Not using a config file.

### What happened?

When running on the code of a VSCode extension, it warned me that the `.postMessage` are missing the `targetOrigin` argument:

```
  × eslint-plugin-unicorn(require-post-message-target-origin): Missing the `targetOrigin` argument.
    ╭─[src/vscode/overview-view.ts:35:45]
 34 │     if (this._view) {
 35 │       this._view.webview.postMessage(message);
    ·                                             ─
 36 │     }
    ╰────
  help: Insert `, self.location.origin`

```

While this is true in the browser, for `window.postMessage`, it is not the case for VSCode's webview `.postMessage` (see https://code.visualstudio.com/api/references/vscode-api#Webview) which only takes a message argument.
I would expect the linter to infer the correct signature and not warn here.

**Comment by shulaoda** [comment]
This behavior aligns with `unicorn`. I think we can use `oxlint-disable` to handle this special case, or we can introduce an `ignore` option to exclude specific callee objects.

**Comment by tmr232** [comment]
Thanks for the quick response!
I suppose in this case I'll disable the rule for the relevant files, thanks!
